canonical-web-and-design/maas-ui
# fix(all): display node tags as names instead of IDs

## Done

- Updated `Tag` type and fixed resulting type errors
- Fixed displaying node tags as their names instead of their IDs
  - Added a hook `useNodeTags` which returns a list of detailed tags for a given node

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Run the full app using `yarn start`
- Check that tags render as names in the following places
  - Machine list
  - Machine details > Summary
  - Device list
  - Device details > Summary
  - Controller details > Summary

## Fixes

Fixes #3511
